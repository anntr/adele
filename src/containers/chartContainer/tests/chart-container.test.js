import React from 'react';
import ChartContainer, { dataAggregator } from '../chart-container';

const data = [{
  uiKit: {
    data: 'no',
    label: 'UI kit',
  },
  storybook: {
    data: 'no',
    url: '',
    label: 'storybook',
  },
  colorNaming: {
    data: [
      'natural (e.g. fill-red)',
      'specific (e.g.fill-facebook)',
    ],
    label: 'color naming',
  },
},
{
  uiKit: {
    data: [
      'Sketch',
      'Photoshop',
    ],
    url: [
      'http://developer.royalcanin.com/resources.html',
      'http://developer.royalcanin.com/resources.html',
    ],
    label: 'UI kit',
  },
  storybook: {
    data: 'no',
    url: '',
    label: 'storybook',
  },
  colorNaming: {
    data: 'natural (e.g. gray darkest)',
    label: 'color naming',
  },
}];

const parsedData = dataAggregator(data);

describe('data aggregator', () => {
  it('should be able to count binary data in column', () => {
    expect(parsedData).toHaveProperty('storybook.no', 2);
  });

  it('should be able to count array data in column', () => {
    expect(parsedData).toHaveProperty('UI kit.no', 1);
    expect(parsedData).toHaveProperty('UI kit.sketch', 1);
    expect(parsedData).toHaveProperty('UI kit.photoshop', 1);
  });

  it('should not contain "system" property', () => {
    expect(parsedData).not.toHaveProperty('system');
  });

  it('should not contain "company" property', () => {
    expect(parsedData).not.toHaveProperty('company');
  });

  it('should correctly handle different examples of color naming', () => {
    expect(parsedData).toHaveProperty('color naming.natural', 2);
    expect(parsedData).toHaveProperty('color naming.specific', 1);
  });
});

it("should render properly", () => {
  const wrapper = mount(<ChartContainer category="storybook" />);
  expect(wrapper).toMatchSnapshot();
})
