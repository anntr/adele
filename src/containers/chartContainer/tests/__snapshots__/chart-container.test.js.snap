// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<ChartContainer
  category="storybook"
>
  <Chart
    title="storybook category summary"
  >
    <styled.div>
      <div
        className="sc-bdVaJa fHgTKa"
      >
        <styled.h3>
          <h3
            className="sc-bwzfXH crWMJl"
          >
            storybook category summary
          </h3>
        </styled.h3>
        <Plot
          data={Array []}
        >
          <FlexibleXYPlot
            margin={
              Object {
                "left": 160,
                "right": 150,
              }
            }
            yType="ordinal"
          >
            <div
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              <XYPlot
                animation={null}
                className=""
                height={0}
                margin={
                  Object {
                    "left": 160,
                    "right": 150,
                  }
                }
                width={0}
                yType="ordinal"
              >
                <div
                  className="rv-xy-plot "
                  style={
                    Object {
                      "height": "0px",
                      "width": "0px",
                    }
                  }
                />
              </XYPlot>
            </div>
          </FlexibleXYPlot>
        </Plot>
      </div>
    </styled.div>
  </Chart>
</ChartContainer>
`;
